tests/test_db_constraints.py

```diff
@@ -0,0 +1,569 @@
+    assert exists is None
+
+    exists = models.User.query.filter_by(username=username).one_or_none()
+    assert exists is not None
+
+
+# Invite ################################################################################## Invite #
+
+
+def __setup_invite(unit_name, invite_email):
+    unit = models.Unit.query.filter_by(name=unit_name).first()
+    invite = models.Invite(email=invite_email, role="Researcher")
+
+    unit.invites.append(invite)
+
+    db.session.add(invite)
+    db.session.commit()
+
+    invite = models.Invite.query.filter_by(email=invite_email).first()
+    assert invite is not None
+    assert invite.unit is not None
+
+    return invite
+
+
+def test_delete_invite(client):
+    """
+    Invite row deleted
+
+        Unit row kept
+    """
+    unit_name = "Unit 1"
+    invite_email = "invite_email@example.com"
+    invite = __setup_invite(unit_name, invite_email)
+
+    db.session.delete(invite)
+    db.session.commit()
+
+    exists = models.Invite.query.filter_by(email=invite_email).first()
+    assert exists is None
+
+    exists = models.Unit.query.filter_by(name=unit_name).first()
+    assert exists is not None
+
+
+# DeletionRequest ################################################################ DeletionRequest #
+
+
+def __setup_deletion_request(username):
+    deletion_request = models.DeletionRequest.query.filter_by(requester_id=username).first()
+    assert deletion_request is not None
+    assert deletion_request.requester is not None
+    return deletion_request
+
+
+def test_delete_deletion_request(client):
+    """
+    DeletionRequest row deleted
+
+        User row kept
+    """
+    username = "researchuser"
+    deletion_request = __setup_deletion_request(username)
+
+    db.session.delete(deletion_request)
+    db.session.commit()
+
+    exists = models.DeletionRequest.query.filter_by(requester_id=username).one_or_none()
+    assert exists is None
+
+    exists = models.User.query.filter_by(username=username).one_or_none()
+    assert exists is not None
+
+
+# File ###################################################################################### File #
+
+
+def __setup_file(filename):
+    file = models.File.query.filter_by(name=filename).first()
+
+    assert file is not None
+    assert file.project is not None
+    assert file.versions != []
+    return file
+
+
+def test_delete_file(client):
+    """
+    File row deleted
+
+        Project row kept
+        File versions kept
+    """
+    filename = "filename1"
+    file = __setup_file(filename)
+
+    project_id = file.project.id
+    version_ids = [version.id for version in file.versions]
+
+    db.session.delete(file)
+    db.session.commit()
+
+    exists = models.File.query.filter_by(name=filename).one_or_none()
+    assert exists is None
+
+    exists = models.Project.query.filter_by(id=project_id).one_or_none()
+    assert exists is not None
+    print(file.id)
```

Comment:
Do we want this here?

Reply:
Just wondering if it's just a debugging thing for now, if it's the error on your laptop again, or if we can remove it

Reply:
Just debugging left by mistake, will fix!